galley/pkg/config/util/pilotadapter/convert.go

```diff
@@ -0,0 +1,174 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilotadapter
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"istio.io/istio/galley/pkg/config/schema/collection"
+	"istio.io/istio/galley/pkg/config/schema/resource"
+	"istio.io/istio/galley/pkg/config/util/pb"
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config/schema"
+	"istio.io/istio/pkg/util/gogoprotomarshal"
+)
+
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertPilotSchemasToGalley(in schema.Set) collection.Schemas {
+	b := collection.NewSchemasBuilder()
+	for _, i := range in {
+		b.MustAdd(ConvertPilotSchemaToGalley(i))
+	}
+	return b.Build()
+}
+
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertPilotSchemaToGalley(in schema.Instance) collection.Schema {
+	return collection.Builder{
+		Name:         in.Collection,
+		VariableName: in.VariableName,
+		Resource: resource.Builder{
+			ClusterScoped: in.ClusterScoped,
+			Kind:          in.Type,
+			Plural:        in.Plural,
+			Group:         in.Group,
+			Version:       in.Version,
+			Proto:         in.MessageName,
+			ValidateProto: in.Validate,
+		}.MustBuild(),
+	}.MustBuild()
+}
+
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertGalleySchemasToPilot(in collection.Schemas) schema.Set {
+	all := in.All()
+	out := make(schema.Set, 0, len(all))
+	for _, s := range all {
+		// Only add known types.
+		if _, ok := crd.KnownTypes[crd.CamelCaseToKebabCase(s.Resource().Kind())]; ok {
+			out = append(out, ConvertGalleySchemaToPilot(s))
+		}
+	}
+	return out
+}
+
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertGalleySchemaToPilot(in collection.Schema) schema.Instance {
+	return schema.Instance{
+		ClusterScoped: in.Resource().IsClusterScoped(),
+		VariableName:  in.VariableName(),
+		Type:          in.Resource().Kind(),
+		Plural:        in.Resource().Plural(),
+		Group:         in.Resource().Group(),
+		Version:       in.Resource().Version(),
+		MessageName:   in.Resource().Proto(),
+		Validate:      in.Resource().ValidateProto,
+		Collection:    in.Name().String(),
+	}
+}
+
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertConfigToObject(schema collection.Schema, cfg model.Config) (crd.IstioObject, error) {
+	spec, err := gogoprotomarshal.ToJSONMap(cfg.Spec)
+	if err != nil {
+		return nil, err
+	}
+	namespace := cfg.Namespace
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
```

Comment:
`if namespace == "" && !schema.ClusterScoped` to support cluster-scoped resources?

Reply:
I think that's right ... I'll make the change. It's not in the original pilot code, however.